### ISSUE TITLE
SEC-2029: j_spring_security_check in https

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-security#2247]
[sreekanth](https://jira.spring.io/secure/ViewProfile.jspa?name=sreekanth.nair@egovernments.org) (Migrated from [SEC-2029](https://jira.spring.io/browse/SEC-2029?redirect=false)) said:

Its totally weird that, when i turn on https for j_spring_security_check, i can't able to login it shows the following security error message

Authentication method not supported: GET

my login page method attribute is post only, when i checked the network log it shows.

```
Request URL:http://localhost:8080/sample-web/j_spring_security_check
Request Method:POST
Status Code:302 Moved Temporarily
Request Headersview source
Accept:text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
Accept-Charset:ISO-8859-1,utf-8;q=0.7,*;q=0.3
Accept-Encoding:gzip,deflate,sdch
Accept-Language:en-US,en;q=0.8
Cache-Control:max-age=0
Connection:keep-alive
Content-Length:31
Content-Type:application/x-www-form-urlencoded
Cookie:JSESSIONID=277928532DA516449D8FC8D4843F4C92
Host:localhost:8080
Origin:http://localhost:8080
Referer:http://localhost:8080/sample-web/login
User-Agent:Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.4 (KHTML, like Gecko) Chrome/22.0.1229.0 Safari/537.4
Form Dataview URL encoded
j_username:xyz
j_password:xyzpass
Response Headersview source
Content-Length:0
Date:Thu, 09 Aug 2012 09:59:38 GMT
Location:https://localhost:8443/sample-web/j_spring_security_check
Server:Apache-Coyote/1.1
```

Basically when i submit my page /j_spring_security_check is saying HTTP status 302.

following is my security context xml's http config part

``` xml
<http use-expressions="true">
        <form-login login-page="/login" default-target-url="/" />
        <http-basic />
        <logout logout-success-url="/login" />
        <intercept-url pattern="/login*" access="isAnonymous()" />
        <intercept-url pattern="/resources/**" access="isAnonymous()"/>
        <intercept-url pattern="/j_spring_security_check" access="isAnonymous()" requires-channel="https" method="POST"/>
        <intercept-url pattern="/collection/**" access="USER" requires-channel="https"/>
        <intercept-url pattern="/**" access="hasAnyRole('USER','ADMIN')" />
    </http>
```
